sphinx/source/docs/dev_guide/bokehjs.rst

```diff
@@ -3,220 +3,310 @@
-the string ``bk-``. For instance some examples are: ``.bk-plot``, ``.bk-toolbar-button``, etc.
+The CSS definitions for BokehJS are contained in several ``.less`` files in the
+:bokeh-tree:`bokehjs/src/less/` directory. All CSS classes for Bokeh DOM
+elements are prefixed with ``bk-``. For example: ``.bk-plot`` or
+``.bk-toolbar-button``.
 
 .. _contributor_guide_bokehjs_development:
+.. _contributor_guide_bokehjs_style_guide:
 
-Development
------------
+Code Style Guide
+-----------------
 
-BokehJS's source code is located in the :bokeh-tree:`bokehjs` directory in Bokeh's
-monorepo repository. All further instructions and shell commands assume that
-``bokehjs/`` is the current directory.
+BokehJS doesn't have an explicit style guide. Make sure to run ``node make
+lint`` or ``node make lint --fix`` before committing to the Bokeh repository.
+Also, review the surrounding code and try to be consistent with the existing
+code.
 
-Some guidelines to adhere to when working on BokehJS:
+Some guidelines and tips to keep in mind when working on BokehJS:
 
 * Do not use ``for-in`` loops, especially unguarded by ``hasOwnProperty()`` Use
   ``for-of`` loop in combination with ``keys()``, ``values()`` and/or
   ``entries()`` from the ``core/util/object`` module instead.
-
-Requirements
-~~~~~~~~~~~~
+* Use double quotes (``"string"``) for strings by default. Use single
+  quotes in cases where they help you avoid escaping quotation marks
+  (``case '"': return "&quot;"``).
+* Use `template literals (template strings)`_ for multiline, tagged, and
+  interpolated strings ( ```Bokeh ${Bokeh.version}``` ).
+
+Always lint your BokehJS code with ESLint_: From the :bokeh-tree:`bokehjs`
+directory, run ``node make lint`` to check your code. Run
+``node make lint --fix`` to have ESLint fix some problems automatically. For
+more details, see the rules defined in :bokeh-tree:`bokehjs/eslint.json`.
+
+.. tip::
+  If you use VSCode, you can use the following configuration for your workspace
+  to use ESLint directly in the editor:
+
+  .. code-block:: json
+
+        "eslint.format.enable": true,
+        "eslint.lintTask.enable": true,
+        "eslint.debug": false,
+        "eslint.quiet": false,
+        "eslint.options": {
+          "cache": true,
+          "extensions": [".ts"],
+          "overrideConfigFile": "./eslint.json"
+        },
+        "eslint.workingDirectories": [
+          "./bokehjs"
+        ]
+
+  This requires the `ESLint extension for VSCode`_ and ESLint version 8 or above
+  to be installed.
+
+.. _contributor_guide_bokehjs_requirements:
+
+Development requirements
+------------------------
+
+To build and test BokehJS locally, follow the instructions in
+:ref:`contributor_guide_setup`. This way, all required packages
+should be installed and configured on your system.
+
+BokehJS requires the following minimum versions:
 
 * node 14+
 * npm 7.4+ (most recent version)
```

Comment:
@mattpap npm 8 seems to be available now, should we update this requirement to that version then?

Reply:
7.4 is the lower bound. We need to test if npm 8 works, e.g. maybe it upgrades lock file format. If that's the case, then we will have to update the lower bound. Otherwise there's no need to force a newer version.